include/xtensor/xbuilder.hpp

```diff
@@ -86,6 +87,93 @@ namespace xt
     }
 #endif
 
-}
-#endif
+    namespace detail
+    {
+
+        template <class Functor, class I>
+        inline auto make_xgenerator(Functor&& f, std::initializer_list<I> shape) noexcept
```

Comment:
This is a detail which could be fixed later (since it is quite cumbersome), but we cannot determine the size of an instance of `initializer_list` at build time, which would prevent having `std::array` shape type. For the case of `broadcast`, we have a static array overload, which has the only issue of not working with old versions of clang.